retention/cohort_analysis.py

```diff
@@ -43,6 +43,7 @@ def get_conn(self):
         conn = connect(
             dbname=self.SYNAPSE_DB_DATABASE,
             user=self.SYNAPSE_DB_USERNAME,
+            port=os.environ.get("SYNAPSE_DB_PORT"),
```

Comment:
Shouldn't this follow the same pattern as the other variables, which are set in the constructor immediately above?

Reply:
well maybe, but that seems entirely redundant. I could refactor the others if you want?

Reply:
Either way, as long as it's consistent :)

Reply:
fair enough.

I ended up refactoring it to build a DSN in the constructor (which is what `connect` does internally if you pass it a load of kwargs). WDYT?